proof-systems/src/darlin/poly-commit/src/constraints/mod.rs

```
@@ -0,0 +1,446 @@
+        commitment: &Self::Commitment,
+        point: &NonNativeFieldGadget<G::ScalarField, ConstraintF>,
+        value: &Vec<Boolean>, // bits of evaluation point in big-endian order
+        proof: &Self::Proof,
+        random_oracle: &mut Self::RandomOracle,
+    ) -> Result<Self::VerifierState, Self::Error>;
+
+    /// succinct check of the verification of an opening proof for multiple polynomials in the
+    /// same point
+    fn succinct_verify_single_point_multi_poly<CS: ConstraintSystemAbstract<ConstraintF>>(
+        mut cs: CS,
+        vk: &Self::VerifierKey,
+        labeled_commitments: &[LabeledCommitmentGadget<Self, ConstraintF, G, PC>], //ToDo: see if we can take an impl IntoIterator as for the primitive
+        point: &NonNativeFieldGadget<G::ScalarField, ConstraintF>,
+        values: &[NonNativeFieldGadget<G::ScalarField, ConstraintF>],
+        proof: &Self::Proof,
+        random_oracle: &mut Self::RandomOracle,
+    ) -> Result<Self::VerifierState, Self::Error> {
+        let lambda = random_oracle.enforce_get_challenge::<_, 128>(
+            cs.ns(|| "squeeze lambda for single-point-multi-poly verify"),
+            )?;
+        let lambda_non_native = Self::challenge_to_non_native_field_element(
+            cs.ns(|| "convert lambda to non native field gadget"),
+            &lambda,
+        )?;
+        let default_commitment = LabeledCommitmentGadget::new(
+            String::from("default commitment"),
+            Self::Commitment::zero(&mut cs)?,
+        );
+        let mut commitments_iter = labeled_commitments.iter().rev();
+        let mut batched_commitment = commitments_iter
+            .next()
+            .unwrap_or(&default_commitment)
+            .commitment()
+            .clone();
+
+        let zero_value = NonNativeFieldGadget::<G::ScalarField, ConstraintF>::zero(&mut cs)?;
+        let mut values_iter = values.iter().rev();
+        let mut batched_value = values_iter.next().unwrap_or(&zero_value).clone();
```

Instead of supplying default values if the user passes an empty list of commitments and/or values (that it's in reality a complete misuse of the API and using 0 values can cause problems), let's throw an error if such lists don't contain at least one element (and perhaps let's also throw an error if they are not of the same length). Please maybe add these checks also to the primitive functions if not already present

---

Added checks for these corner cases. I added also similar checks to the `succinct_verify_multi_poly_multi_point` function